R/R6Wrappers.R

```diff
@@ -114,58 +147,146 @@ createSection <- function(name, ordinal, lineItems) {
+                                           domain = domain,
+                                           conceptSet = conceptSet,
+                                           timeInterval = timeInterval,
                                            sourceConceptSet = sourceConceptSet,
                                            typeConceptIds = typeConceptIds,
                                            visitOccurrenceConceptIds = visitOccurrenceConceptIds)
   return(csDefinition)
 }
 
+# function to get timeInterval and concept set combinations
+.permuteCsTi <- function(conceptSets, timeIntervals) {
+
+  # get number of items for each
+  numCs <- length(conceptSets)
+  numTis <- length(timeIntervals)
+
+  # build out permutations
+  csPerm <- rep(conceptSets, times = numTis)
+  tiPerm <- rep(timeIntervals, each = numCs)
+
+  permTiCs <- list(
+    'conceptSets' = csPerm,
+    'timeIntervals' = tiPerm
+  )
+  return(permTiCs)
+
+}
+
 #' @title
-#' Create a batch of concept set line items from a list of Capr concept sets. 
-#' 
-#' @description 
-#' The name of each line item will be set to the name of its Capr concept set, and the ordinal will be set to the index of the Capr concept set in the list. All line items will use the same statistic, domain, type concepts, and visit concepts. It is not possible to specify source concept IDs.
+#' Create a batch of concept set line items from a list of Capr concept sets.
 #'
+#' @description
+#' The name of each line item will be set to the name of its Capr concept set, and the ordinal will be set to the index of the Capr concept set in the list. All line items will use the same statistic, domain, type concepts, and visit concepts. It is not possible to specify source concept IDs.
+#' @param name The name of the concept set batch
 #' @param statistic The Statistic object to be used to evaluate the line items
-#' @param conceptSets A list of concept set Capr objects
 #' @param domain The domain of the concept sets (must be one of 'Condition', 'Drug', 'Procedure', 'Observation', 'Measurement', 'Device')
+#' @param conceptSets A list of concept set Capr objects
+#' @param timeIntervals A list of TimeInterval class objects
+#' @param sourceConceptSet (OPTIONAL) A Capr concept set of source concept IDs to use to limit the concept set
 #' @param typeConceptIds (OPTIONAL) A list of type concept IDs to use to limit the concept set
 #' @param visitOccurrenceConceptIds (OPTIONAL) A list of visit occurrence concept IDs to use to limit the concept set
 #'
 #' @return A list of ConceptSetDefinition objects
 #'
 #' @export
-createConceptSetLineItemBatch <- function(statistic,
-                                          conceptSets,
-                                          domain,
-                                          typeConceptIds = c(),
-                                          visitOccurrenceConceptIds = c()) {
+createConceptSetLineItemBatch <- function(
```

Comment:
can you please also add a helper function that creates a batch of line items for a single concept set across multiple time windows? assume this will be a common use case

Reply:
already handled with the permutation operator